src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -0,0 +1,1236 @@
+# coding=utf-8
+# Copyright 2022 The Fairseq Authors and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Whisper model."""
+
+
+import math
+import random
+from typing import Optional, Tuple
+
+import torch
+import torch.nn.functional as F
```

Comment:
(nit) we actually had a PR that replaced all one-letter imports by just using `nn.functional` directly, so feel free to do it here as well